packages/plugins/typescript/effect/src/visitor.ts

```diff
@@ -0,0 +1,198 @@
+import autoBind from 'auto-bind';
+import { type GraphQLSchema, type OperationDefinitionNode, print } from 'graphql';
+import {
+  type ClientSideBasePluginConfig,
+  ClientSideBaseVisitor,
+  DocumentMode,
+  LoadedFragment,
+  type RawClientSideBasePluginConfig,
+} from '@graphql-codegen/visitor-plugin-common';
+
+export interface EffectPluginConfig extends ClientSideBasePluginConfig {}
+
+const additionalStaticContent = (documentMode: DocumentMode) => `
+export type GraphQLSuccessResponse<A> = {
+  body: ExecutionResult & { data: A };
+  headers: Record<string, string>;
+};
+
+export type GraphQLErrorResponse = {
+  body: Omit<ExecutionResult, 'data'>;
+  headers: Record<string, string>;
+};
+
+export class MissingDataGraphQLResponseError extends Data.TaggedError(
+  'MissingDataGraphQLResponseError',
+)<GraphQLErrorResponse> {}
+
+export type GraphQLOperationOptions = {
+  preferredOpName?: string;
+};
+
+type GraphQLOperationArgs = {
+  document: ${documentMode === DocumentMode.string ? 'string' : 'DocumentNode'};
+  fallbackOperationName: string;
+};
+
+// https://github.com/graphql/graphql-over-http/blob/main/spec/GraphQLOverHTTP.md#legacy-watershed
+const Accept = 'application/graphql-response+json; charset=utf-8, application/json; charset=utf-8';
+
+const makeGraphQLClient = Http.client.mapRequest(
+  Http.request.setHeaders({ Accept, 'Content-Type': 'application/json' }),
+);
+
+const makeGraphQLOperation =
+  <Vars, Data>({ document, fallbackOperationName }: GraphQLOperationArgs) =>
+  (variables: Vars, opts?: GraphQLOperationOptions) => {
+    const operationName = opts?.preferredOpName ?? fallbackOperationName;
+    const query = ${documentMode === DocumentMode.string ? 'document' : 'print(document)'};
+
+    return Effect.flatMap(Http.client.Client, client =>
+      Http.request.post('').pipe(
+        Http.request.jsonBody({
+          query,
+          operationName,
+          variables,
+        }),
+        Effect.flatMap(Http.client.filterStatusOk(makeGraphQLClient(client))),
```

Comment:
You can move filterStatusOk to `makeGraphQLClient`.

Reply:
Addressed in https://github.com/dotansimha/graphql-code-generator-community/pull/665/commits/185ccb8a138f947fd0ffe330cef63f9e888f4de3.